src/nlsolve/anderson.jl

```diff
@@ -43,15 +45,17 @@ function (S::NLAnderson{false,<:NLSolverCache})(integrator)
     gs[1] = z₊
     
     mk = min(S.n, iter-1)
-    residuals[1:mk] = (gs[2:mk+1] .- zs[2:mk+1]) .- (gs[1] - zs[1])
+    residuals[1:mk] .= (gs[2:mk+1] .- zs[2:mk+1]) .- (gs[1] - zs[1])
     alphas[1:mk] .= residuals[1:mk] \ [(zs[1] - gs[1])]
```

Comment:
```suggestion
    alphas[1:mk] .= residuals[1:mk] \ (zs[1] - gs[1])
```

Reply:
Currently the out-of place version is limited to scalars only thats why I did so. I will make a seperate PR for introducing broadcast to oop method.